packages/components/pie-notification/src/index.ts

```diff
@@ -160,11 +170,9 @@ export class PieNotification extends LitElement implements NotificationProps {
      * @private
      */
     private renderIcon (): TemplateResult | typeof nothing {
-        const hasExternalIcon = this._iconSlot.length > 0;
-
         return html`
-                ${!hasExternalIcon ? this.getDefaultVariantIcon() : nothing}
-                <slot name="icon"></slot>
+                ${!this.hasExternalIcon ? this.getDefaultVariantIcon() : nothing}
+                <slot @slotchange=${this.handleSlotchange} name="icon"></slot>
```

Comment:
I've just realised we might be able to do something different and much simpler...
https://lit.dev/docs/components/shadow-dom/#fallback

I think it's worth trying before we merge this.

```suggestion
                <slot name="icon">${this.getDefaultVariantIcon()}</slot>
```

Reply:
looks like it works, great finding 🔥